backend/main.py

```
@@ -0,0 +1,312 @@
+                    'valence': features['valence'],
+                    'tempo': features['tempo'],
+                    'acousticness': features['acousticness'],
+                    'instrumentalness': features['instrumentalness']
+                }
+            )
+            songs.append(song)
+        
+        return songs
+    except Exception as e:
+        logger.error(f"Spotify API error: {e}")
+        raise HTTPException(status_code=500, detail="Spotify API error")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)